lib/persistence/abstract.js

```diff
@@ -68,7 +68,11 @@ AbstractPersistence.prototype.wire = function(server) {
       total++;
       that.storeRetained(packet, done);
     }
-    that.storeOfflinePacket(packet, done);
+    if (packet.qos !== 0 || that.options.storeMessagesQos0) {
+      total++;
+      that.storeOfflinePacket(packet, done);
+    }
+    done();
```

Comment:
total is initialized to 1, and I've incremented it when storeOfflinePacketing (as like storeRertained behavior), so I believe we should call it always (to simulate already called one by old storeOfflinePacket ).
Am I wrong?